pkg/storages/splitmerge/channelreader.go

```diff
@@ -0,0 +1,53 @@
+package splitmerge
```

Comment:
As I can see, this functional is not storages-specific, why need to put it in `storages`? `internal/splitmerge` or `internal/ioextensions/splitmerge` looks like a good destination.

Reply:
+1